avaje-record-builder-core/src/main/java/io/avaje/recordbuilder/internal/Templates.java

```diff
@@ -20,43 +20,52 @@ static String classTemplate(
 
     return MessageFormat.format(
         """
-		   package {0};
+		   {0}
 
 		   {1}
 
 		   /** Builder class for '{'@link {2}'}' */
 		   @Generated("avaje-record-builder")
-		   public class {2}Builder{8} '{'
+		   public class {2}Builder{8} %s'{'
+
 		   {3}
-		     private {2}Builder() '{'
-		     '}'
 
-		     private {2}Builder({4}) '{'
-		       {5}
+		     private {2}Builder() '{'
 		     '}'
+		   """.formatted(implementsStr)
+            + constructor(constructor)
+            + """
 
 		     /**
 		      * Return a new builder with all fields set to default Java values
 		      */
 		     public static{9}{2}Builder{10} builder() '{'
-		         return new {2}Builder{10}();
+		       return new {2}Builder{10}();
 		     '}'
 
 		     /**
 		      * Return a new builder with all fields set to the values taken from the given record instance
 		      */
 		     public static{9}{2}Builder{10} builder({2}{10} from) '{'
-		         return new {2}Builder{10}({6});
+		       return new {2}Builder{10}({6});
 		     '}'
 
 		     /**
 		      * Return a new {2} instance with all fields set to the current values in this builder
 		      */
 		     public {2}{10} build() '{'
-		         return new {2}{10}({7});
+		       return new {2}{10}({7});
+		     '}'
+
+		     private static <T> T requireNonNull(@Nullable T obj) '{'
+		       if (obj == null) '{'
+		         throw new IllegalStateException(\"{2}Builder expects all nonnull values to not be null \");
```

Comment:
The only remark I have is that this error message is a non-issue in any null-analysis context, _however_, in the default operation (where null can propagate freely) it is difficult to debug:

```
jshell> NullityBuilder.builder().build()
|  Exception java.lang.IllegalStateException: NullityBuilder expects all nonnull values to not be null
|        at NullityBuilder.requireNonNull (#5:38)
|        at NullityBuilder.build (#5:33)
|        at (#7:1)
```

There can be multiple `requireNonNull` statements in the same line so the line number in the stack trace is not much of a help. Sadly I can think of no easy or gratis solve for that -- only passing in the name of the field.

Reply:
> only passing in the name of the field.

hey that's a pretty good idea